calico/etcddriver/driver.py

```diff
@@ -512,11 +532,18 @@ def _handle_next_watcher_event(self, resync_in_progress):
         :raises FelixWriteFailed:
         :raises ResyncRequested:
         """
+        # PLW: What happens to the watcher thread if stop() gets called? Or does
+        # it plough on happily until the process goes down?
```

Comment:
The watcher checks the stop event on each loop.  It might take 90s to die due to the etcd poll timeout.